rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/HistoryTransformers.scala

```diff
@@ -0,0 +1,193 @@
+package coop.rchain.rspace.nextgenrspace.history
```

Comment:
do you feel this is the right path?
in my mind the code should not change how the original execution behaves. in particular there is no way to disable the transformation when debug is disabled. IOW we're introducing a performance hit in production code which should not be there if we are not measuring sizes.